ci-operator/step-registry/cucushift/hypershift-extended/idp-htpasswd/cucushift-hypershift-extended-idp-htpasswd-commands.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+export LC_CTYPE=C
+set -o nounset
+set -o errexit
+set -o pipefail
+
+export KUBECONFIG="${SHARED_DIR}/kubeconfig"
+count=$(oc get hostedclusters --no-headers --ignore-not-found -n clusters | wc -l)
+echo "$count"
+if [ "$count" -lt 1 ]  ; then
+    echo "namespace clusters don't have hostedcluster"
+    exit 1
+fi
+cluster_name=$(oc get hostedclusters -n clusters -o jsonpath='{.items[0].metadata.name}')
+
+# prepare users
+users=""
+htpass_file=/tmp/users.htpasswd
+
+for i in $(seq 1 40);
+do
+    username="testuser-${i}"
+    password=$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true)
+    users+="${username}:${password},"
+    if [ -f "${htpass_file}" ]; then
+        htpasswd -B -b ${htpass_file} "${username}" "${password}"
+    else
+        htpasswd -c -B -b ${htpass_file} "${username}" "${password}"
+    fi
+done
+
+## add users to cluster
+oc create secret generic "$cluster_name" --from-file=htpasswd="$htpass_file" -n clusters
+oc patch hostedclusters $cluster_name -n clusters --type=merge -p '{"spec":{"configuration":{"oauth":{"identityProviders":[{"htpasswd":{"fileData":{"name":"'$cluster_name'"}},"mappingMethod":"claim","name":"htpasswd","type":"HTPasswd"}]}}}}'
+
+# store users in a shared file
+if [ -f "${SHARED_DIR}/runtime_env" ] ; then
+    source "${SHARED_DIR}/runtime_env"
```

Comment:
suggest put all the similar codes together special for [cucushift-pre-commands.sh](https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/pre/cucushift-pre-commands.sh) to make it clear.

Another suggestion:  In `cucushift-pre-commands.sh` it just add the new users into the existing idp oauth users instead of removing the existing users.  I'd like to follow their rules to enable all the users in the management cluster...